transformer_engine/pytorch/cpu_offload.py

```diff
@@ -310,25 +320,33 @@ def get_tensor_buf_for_offloaded_tensor(self, tensor, tensor_tag):
 
 
     def tensor_push(self, tensor: torch.Tensor, **kwargs) -> Any:
-        # obtain a unique tensor tag
-        tensor_tag = (self.current_group, self.tensor_count_current_group)
-        self.tensor_count_current_group += 1
-        assert tensor_tag not in self.tensor_tag_to_state
-
-        if (self.current_group < self.num_offload_group
-            and self.tensor_need_offloading_checker(tensor)):
-            # first copy the tensor to tensorbuf, so that the original tensor will not be deleted
-            tensor_buf = self.get_tensor_buf_for_offloaded_tensor(tensor, tensor_tag)
-            tensor_buf.copy_(tensor)
-            if hasattr(tensor,"weight_offloading"):
-                tensor_buf.weight_offloading = True
-            if hasattr(tensor,"activation_offloading"):
-                tensor_buf.activation_offloading = True
-           # Here we just save it, and at commit, bulk_offload_group will handle it
-            self.tensor_tag_to_state[tensor_tag] = tensor_buf
+        torch_stray_tensor = (type(tensor) != torch.Tensor) and check_torch_stray_tensor
```

Comment:
When using torch.compile tensor types like FunctionalTensor and FakeTensors flow into this function which we want to avoid. Hence this check.

Reply:
Ok, but you are still saving them in line 347, so I'm a little confused here. Could you explain what exactly is this function supposed to do?

Reply:
Well, they don't need to be revived because you store the actual object so no need to revive anything?

Reply:
So what I mean by revive is autograd doing an implicit ctx.saved_tensors in backprop. For some reason (which only Torch developers know why), not all these Functional/Fake tensors are revived back during autograd, only a few are. So, assertion check fails for the one's that are not revived, hence this way of avoiding the check.